src/main/java/de/medizininformatikinitiative/fhir_data_evaluator/DataStore.java

```diff
@@ -24,21 +24,21 @@
 @Component
 public class DataStore {
 
-    private final WebClient client;
+    private final WebClient sourceClient;
+    private final WebClient destinationClient;
     private final IParser parser;
-    private final int pageCount;
-    private final URI reportDestinationServer;
+    private final int sourcePageCount;
     private final FhirContext context;
     private final IFhirPath applicationFhirPathEngine;
 
-    public DataStore(WebClient client, IParser parser, @Value("${fhir.pageCount}") int pageCount,
-                     @Value("${fhir.reportDestinationServer}") URI reportDestinationServer,
+    public DataStore(WebClient sourceClient, WebClient destinationClient, IParser parser,
```

Comment:
The DataStore should only address one FHIR server. Please use a second instance of the Datastore for the report FHIR server.

Reply:
Do you mean i should create another class declared as Component again, or shouild i keep it in this single class and just declare it not as Component but as a regular class and then create two beans from it?

Reply:
Single class und two instances/beans. The DataStore should be like an SDK for a FHIR API. At least the parts we need.